components/mfa/setup/setup.tsx

```diff
@@ -3,34 +3,59 @@
 
 import React from 'react';
 import {FormattedMessage} from 'react-intl';
-import PropTypes from 'prop-types';
+import {UserProfile} from 'mattermost-redux/src/types/users';
 
 import * as Utils from 'utils/utils.jsx';
 import {t} from 'utils/i18n.jsx';
 
 import FormattedMarkdownMessage from 'components/formatted_markdown_message.jsx';
 import LocalizedInput from 'components/localized_input/localized_input';
 
-export default class Setup extends React.PureComponent {
-    static propTypes = {
-        currentUser: PropTypes.object,
-        siteName: PropTypes.string,
-        enforceMultifactorAuthentication: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            activateMfa: PropTypes.func.isRequired,
-            generateMfaSecret: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    currentUser: UserProfile;
+    siteName?: string;
+    enforceMultifactorAuthentication: boolean;
+    actions: {
+        activateMfa: (code: string) => Promise<{
+            error: {
+                server_error_id: string;
+                message: string;
+            }
+        }>;
+        generateMfaSecret: () => Promise<{
+            data: {
+                secret: string;
+                qr_code: string;
+            };
+            error: {
```

Comment:
@kashifsoofi Looking at the error message, it looks like the type you have defined here doesn't match the type provided by the redux connector. In the redux connector, the type on `error` is optional, whereas here it is not.